Source/Private/ASDisplayNodeTipState.h

```diff
@@ -0,0 +1,31 @@
+//
+//  ASDisplayNodeTipState.h
+//  AsyncDisplayKit
+//
+//  Created by Adlai Holler on 4/12/17.
+//  Copyright © 2017 Facebook. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import <AsyncDisplayKit/ASBaseDefines.h>
+
+@class ASDisplayNode, ASTipNode;
+
+NS_ASSUME_NONNULL_BEGIN
+
+AS_SUBCLASSING_RESTRICTED
+@interface ASDisplayNodeTipState : NSObject
+
+- (instancetype)initWithNode:(ASDisplayNode *)node NS_DESIGNATED_INITIALIZER;
+
+- (instancetype)init NS_UNAVAILABLE;
+
+/// Unsafe because once the node is deallocated, we will not be able to access the tip state.
+@property (nonatomic, unsafe_unretained, readonly) ASDisplayNode *node;
```

Comment:
So that the `node` property is truly immutable. If we keep tip states around after nodes are deallocated, then we should crash so that we stop doing that. This acts like an assertion, similar to Swift's `unowned` storage class.